src/ORM/Marshaller.php

```diff
@@ -105,6 +105,22 @@ public function one(array $data, array $options = [])
         $propertyMap = $this->_buildPropertyMap($options);
 
         $schema = $this->_table->schema();
+        $primaryKey = $schema->primaryKey();
+
+        if (array_intersect($primaryKey, array_keys($data)) == $primaryKey) {
+            $tableName = $this->_table->alias();
+            $record = $this->_table->find('all');
+            foreach ($primaryKey as $pkey) {
+                $record->where(["$tableName.$pkey" => $data[$pkey]]);
```

Comment:
:+1: For now I have an "AppTable" with this function cause I often need it for my custom finders.